### ISSUE TITLE
Should `lint_package()` error on CI?

### ISSUE THREAD

**Issue by hadley** [r-lib/lintr#1226]
Currently, https://github.com/r-lib/lintr/blob/main/.github/workflows/lint.yaml doesn't actually do anything because `lint_package()` never errors. Should it?  Or should there be another function that's designed specifically for CI scenarios?

**Comment by hadley** [comment]
Markdown report could just be third option in `print.lints`

**Comment by AshesITR** [comment]
We currently produce check annotations, and support error_on_lint to exit if a lint is found.
Maybe we can default `error_on_lint` to `in_ci()`?

**Comment by MichaelChirico** [comment]
@hadley PTAL at this repo's current GHA set-up. We recently updated `lint-changed-files` so that the action fails if new lints are detected. That's working pretty well for us:

https://github.com/r-lib/lintr/blob/bed54c55523e96f7c89154ae31702cd74bb209d8/.github/workflows/lint-changed-files.yaml#L33-L39

Please help to close if you agree that's a good approach

**Comment by hadley** [comment]
Yeah, that looks like a good pattern. Do you plan to wrap that up a bit more into a reusable function?

**Comment by MichaelChirico** [comment]
do you have in mind an exported lint_repo() function? or an argument to lint_package()/lint_dir()

my first instinct was just better documentation (esp. if our workflow is added as a template in r-lib/actions)

**Comment by hadley** [comment]
The advantage of making it a function is that once it gets baked into `r-lib/actions` it's easier for you to change the behaviour if you discover there's a better approach.
